sources/Inputs/src/Configuration.cpp

```diff
@@ -41,6 +42,28 @@ updateValue(T& value, const boost::property_tree::ptree& tree, const std::string
     value = value_opt->get_value<T>();
   }
 }
+
+/**
+ * @brief Helper function to update the active power compensation parameter
+ *
+ * @param activePowerCompensation the value to update
+ * @param tree the element of the boost tree
+ */
+static void
+updateActivePowerCompensationValue(Configuration::ActivePowerCompensation& activePowerCompensation, const boost::property_tree::ptree& tree) {
+  std::map<std::string, Configuration::ActivePowerCompensation> enumResolver = {{"P", Configuration::ActivePowerCompensation::P},
+                                                                                {"targetP", Configuration::ActivePowerCompensation::TARGET_P},
+                                                                                {"PMax", Configuration::ActivePowerCompensation::PMAX}};
+  std::string apcString;
+  helper::updateValue(apcString, tree, "ActivePowerCompensation");
+  auto it = enumResolver.find(apcString);
+  if (it != enumResolver.end()) {
+    activePowerCompensation = it->second;
+  } else if (!apcString.empty()) {
+    //LOG(warn) << MESS(BadActivePowerCompensation) << LOG_ENDL; //This doesn't work, I think it always expect args
```

Comment:
It should be this way instead of raw string comment for warn logs. I guess that MESS macro doesn't support having no argument so this is why you uses raw string. Maybe use a message like:
"ActivePowerCompensation was given with the unsupported value %1%. Supported values are: PMAX, P or targetP. Default value PMAX will be used"